gateway-js/src/index.ts

```diff
@@ -1206,6 +1372,17 @@ export class ApolloGateway implements GraphQLService {
   // schema polling). Can be called multiple times safely. Once it (async)
   // returns, all gateway background activity will be finished.
   public async stop() {
+    await Promise.all(
```

Comment:
I think this should be done as part of the switch before?  maybe inside a function/method that is called in a few different phases, or breaking the switch into two switches or something.  but after doing the error checks and `stopping` handling etc.

Reply:
Makes sense, only running this in the `loaded`, `polling`, and `waiting to poll` states now.
36f9dedf